src/app/collective-rewards/allocations/hooks/useAllocateVotes.ts

```diff
@@ -0,0 +1,58 @@
+import { useWaitForTransactionReceipt, useWriteContract } from 'wagmi'
+import { useAwaitedTxReporting } from '@/app/collective-rewards/shared/hooks'
+import { BackersManagerAbi } from '@/lib/abis/v2/BackersManagerAbi'
+import { BackersManagerAddress } from '@/lib/contracts'
+import { useContext } from 'react'
+import { AllocationsContext } from '../context'
+import { getVoteAllocations } from '../context/utils'
+
+export const useAllocateVotes = () => {
+  const { writeContractAsync, error: executionError, data: hash, isPending } = useWriteContract()
+
+  const {
+    initialState: { allocations: initialAllocations },
+    state: { allocations, getBuilder, isValidState },
+  } = useContext(AllocationsContext)
+
+  const { isLoading, isSuccess, data, error: receiptError } = useWaitForTransactionReceipt({ hash })
+
+  const error = executionError ?? receiptError
+
+  const saveAllocations = () => {
+    const [gauges, allocs] = getVoteAllocations({
+      initialAllocations,
+      currentAllocations: allocations,
+      getBuilder,
+    })
+
+    const isBatchAllocation = allocs.length > 1
```

Comment:
What if `allocs.length === 0`? I think atm the function will fail, so I think we should just return in that case.